File: planet-examples/src/main/java/org/saar/minecraft/IChunk.java
package org.saar.minecraft;

public interface IChunk {

    int getHeight(int x, int z);

    int getSolidHeight(int x, int z);

    int getLight(int x, int y, int z);

    Block getBlock(int x, int y, int z);

    void setBlock(int x, int y, int z, Block block);

    void updateMesh(World world);
}

File: planet-examples/src/main/java/org/saar/minecraft/IChunk.kt
package org.saar.minecraft

interface IChunk {
    fun getHeight(x: Int, z: Int): Int

    fun getLight(x: Int, y: Int, z: Int): Int

    fun getBlock(x: Int, y: Int, z: Int): Block

    fun setBlock(x: Int, y: Int, z: Int, block: Block)

    fun updateMesh()
}
